prime-router/docs/proposals/0008-cdc-ip-space.md

```diff
@@ -0,0 +1,132 @@
+| private | VNET-only; not Internet-routable | 10.1.3.0/24 | 256 | 5 + 0 = 5 | 251 |
+| endpoint | [Azure Private Endpoints](https://docs.microsoft.com/en-us/azure/private-link/private-endpoint-overview) | 10.1.5.0/24 | 256 | 5 + 1 = 6 | 250 |
+
+#### VNET Summary
+
+| Count | Label |
+|--:|:--|
+| 65,536 | IPs in CIDR |
+| 512 | Allocated IPs to Subnets |
+| 65,024 | Available IPs |
+
+## What Will Change with a CDC VNET
+
+Looking toward the future, there will be subnets we will not utilize when we migrate to a CDC VNET:
+
+* **‌GatewaySubnet**
+    * We will no longer be able to manage our own VPN, as the Palo Alto firewall will not allow ingress into the VNET
+        * Ingress is allowed through CyberArk, which will be adopting in the future
+    * We will be able to use the Azure Console, CLI, and Terraform to manage resources, as Azure resources connect both to the VNET and the Internet
+        * We will, however, need to manually manage IP rules until CyberArk is deployed
+* **endpoint**
+    * Since certain Azure resources sever their connection to the internet when a private endpoint is deployed (ex. function app), without a VPN, we will no longer be able to use private endpoints
+
+## Proposed IP CIDR
+
+With the above considerations, the proposed CIDR request per environment/region pair is:
+
+**x.x.x.x/23**
+
+### Why a `/23`?
+
+* A `/23` will grant us 512 IP addresses per VNET
+* While we currently use a limited number of IP addresses, we cannot anticipate our future needs
+* We may want to utilize private endpoints again in the future
+* While our current utilized resources do not consume VNET ip addresses when scaled (ex. function app, database, etc), we may in the future have bare containers that need to be scaled, which will consume an IP per server
+
+### Why request a `/23` in each environment-region pair?
+
+* Environments should be identical
+    * When we conduct full-scale load testing in a lower environment, we’ll want the same number of IPs to verify our scaling logic
+* Regions should allow for full redundant replication
+    * While the only resource currently deployed in our secondary region is a database replica, we’ll want the capability to stand up our entire system in our secondary region
+    * This would grant us full region redundancy and/or provide a closer network path for users on the west coast
+
+#### Subnet Usage
+
+| Subnet | Description | CIDR | IPs | Used | Available |
+|:--|:--|--:|--:|--:|--:|
+| public | Internet-routable resources | x.x.x.x/26 | 64 | 5 + 0 = 5 | 59 |
+| container | Docker containers (ex. SFTP test) | x.x.x.x/26 | 64 | 5 + 2 = 7 | 57 |
+| private | VNET-only; not Internet-routable | x.x.x.x/26 | 64 | 5 + 0 = 5 | 59 |
+| GatewaySubnet | Not used under a CDC VNET |  | |  |  |
+| endpoint | Not used under a CDC VNET |  | |  |  |
+
+#### VNET Summary
+
+| Count | Label |
+|--:|:--|
+| 512 | IPs in CIDR |
+| 192 | Allocated IPs to Subnets |
+| 320 | Available IPs |
```

Comment:
@jimduff-usds That's a fair statement that we're requesting too many IPs. Thoughts on 128 IPs per VNET?

```suggestion
**x.x.x.x/25**

### Why a `/25`?

* A `/25` will grant us 128 IP addresses per VNET
* While we currently use a limited number of IP addresses, we cannot anticipate our future needs
* We may want to utilize private endpoints again in the future
* While our current utilized resources do not consume VNET ip addresses when scaled (ex. function app, database, etc), we may in the future have bare containers that need to be scaled, which will consume an IP per server

### Why request a `/25` in each environment-region pair?

* Environments should be identical
    * When we conduct full-scale load testing in a lower environment, we’ll want the same number of IPs to verify our scaling logic
* Regions should allow for full redundant replication
    * While the only resource currently deployed in our secondary region is a database replica, we’ll want the capability to stand up our entire system in our secondary region
    * This would grant us full region redundancy and/or provide a closer network path for users on the west coast

#### Subnet Usage

| Subnet | Description | CIDR | IPs | Used | Available |
|:--|:--|--:|--:|--:|--:|
| public | Internet-routable resources | x.x.x.x/28 | 16 | 5 + 0 = 5 | 11 |
| container | Docker containers (ex. SFTP test) | x.x.x.x/28 | 16 | 5 + 2 = 7 | 9 |
| private | VNET-only; not Internet-routable | x.x.x.x/28 | 16 | 5 + 0 = 5 | 11 |
| GatewaySubnet | Not used under a CDC VNET |  | |  |  |
| endpoint | Not used under a CDC VNET |  | |  |  |

#### VNET Summary

| Count | Label |
|--:|:--|
| 128 | IPs in CIDR |
| 48 | Allocated IPs to Subnets |
| 80 | Available IPs |
```